docs/zkapps/front-end-integration-guides/angular.mdx

```diff
@@ -0,0 +1,419 @@
+- Verify that "Compiled Add" is logged, meaning that the SDK has successfully generated a proving key for the `Add` zkapp.
+- If you're connected to Devnet and your account is funded with Devnet tokens, you should be be able to broadcast a successful transaction calling `update` on the zkapp. After a few minutes, the state change associated with the transaction will take effect on chain and you'll see `num` increase when you reload the page!
+
+## Deploying to GitHub Pages
+
+- Now we'll set the app up for deployment to GitHub pages.
+- Publish your project to a GitHub repository.
+- Run `ng deploy` and select GitHub Pages.
+- Add `baseHref` to `options` under `build` in angular.json with the name of your GitHub repository.
+    - **Do not remove the slashes!**
+
+```json
+"baseHref": "/<your-github-repo-name>/"
+```
+
+- Create a deploy script in package.json which copies the required libraries
+
+```json
+"deploy": "npm run copy-libs && ng deploy --dir=dist/<project-name>"
+```
+
+- Deploy the app.
+
+```bash
+npm run deploy
+```
+
+- You can view deployment details at `https://github.com/<username>/<repo-name>/actions` and your live site at `https://<username>.github.io/<repo-name>/`.
+
+## SharedArrayBuffer in Deployed Instance
+
+- View the site and open the browser console. You'll see the same error about the SharedArrayBuffer from before! The headers set previously apply only to `ng serve`, so we’ll set them up for GitHub Pages.
+- Install `coi-serviceworker`.
+
+```bash
+npm i coi-serviceworker
+```
+
+- Update the script that copies `o1js` to `public` to also include the `coi-serviceworker` file:
+
+```json
+"copy-libs": "mkdir -p public/lib/o1js && cp node_modules/o1js/dist/web/index.js public/lib/o1js/index.js && cp node_modules/coi-serviceworker/coi-serviceworker.min.js public/lib/coi-serviceworker.min.js"
+```
+
+- Create a file `src/app/COIServiceWorker.ts` with the following contents.
+    - Be sure to replace `<your repo name>` with the name of your GitHub repo!
+
+```tsx
+export {};
+
+function loadCOIServiceWorker() {
+  console.log('Load COIServiceWorker', navigator);
+  if (typeof window !== 'undefined' && 'serviceWorker' in navigator && window.location.hostname != 'localhost') {
+    navigator.serviceWorker.register('/<your repo name>/lib/coi-serviceworker.min.js')
+      .then(registration => console.log('COI Service Worker registered with scope:', registration.scope))
+      .catch(error => console.error('COI Service Worker registration failed:', error));
+  }
+}
+
+loadCOIServiceWorker();
+```
+
+- Import it at the top of `src/app/app.component.ts`.
+
+```tsx
+import './COIServiceWorker';
+```
+
+- Redeploy the application with the `COIServiceWorker` files.
+
+```bash
+npm run deploy
+```
+
+## Congratulations, you’ve developed and deployed a zkApp UI with Angular!
```

Comment:
I still get the post message error on https://45930.github.io/ng-integration-guide/

My COI service worker is registered:
![image](https://github.com/user-attachments/assets/0ac3dff3-440b-4554-bad4-2bfd2b29b5ab)


Do we also need to configure what headers to add?

Reply:
@hattyhattington17 This is an outstanding issue I still have after following the guide.  Most of the other comments are more opinion-based, but this should be resolved.